src/MovieCard.jsx

```diff
@@ -0,0 +1,128 @@
+            setLikeCount(likeCount + 1);
+            heart.innerText = `❤️`;
+        } else if (likeCount >= 1) {
+            const heart = document.getElementById(props.id).querySelector('#like-count');
+            setLikeCount(likeCount - 1);
+            heart.innerText = `♡`;
+        }
+    }
+
+
+    const fetchMoreInfo = async () => {
+        const url = `https://api.themoviedb.org/3/movie/${props.id}?language=en-US&api_key=0d7613c1b95dbc61f3dd491c8f802475`
+        const options = {
+            method: 'GET',
+            headers: {
+              accept: 'application/json',
+              Authorization: `Bearer ${apiKey}`,
+            }
+          };
+
+          const response = await fetch(url, options);
+          const jsonResponse = await response.json();
+          setBackdrop(jsonResponse.backdrop_path);
+          setGenres(jsonResponse.genres);
+          setRuntime(jsonResponse.runtime);
+    }
+
+    const fetchVideo = async () => {
+        const url = `https://api.themoviedb.org/3/movie/${props.id}/videos?language=en-US&api_key=0d7613c1b95dbc61f3dd491c8f802475`
+        const options = {
+            method: 'GET',
+            headers: {
+              accept: 'application/json',
+              Authorization: `Bearer ${apiKey}`,
+            }
+        };
+
+        const response = await fetch(url, options);
+        const jsonResponse = await response.json();
+        const results = jsonResponse.results;
+        let videoKey;
+        results.forEach(item => {
+            videoKey = item.key;
+        });
+        setVideoLink(`https://www.youtube.com/embed/${videoKey}`);
+    }
+
+    function handleLike() {
+        increaseLikes();
+        toggleLike();
+        if (isLiked) {
+            if (props.likedList.some(movie => movie.id === props.id)) {
+                const updatedLikedList = props.likedList.filter(movie => movie.id !== props.id);
+                props.setLikedList(updatedLikedList);
+            }
+        } else {
+            if (!props.likedList.some(movie => movie.id === props.id)) {
+                props.setLikedList(isLikedList => [...isLikedList, { id: props.id, image: props.image, title: props.title }]);
+            }
+        }
+    }
+
+    function handleWatched() {
+        toggleWatched();
+        if (isChecked) {
+            if (props.watchedList.some(movie => movie.id === props.id)) {
+                const updatedWatchedList = props.watchedList.filter(movie => movie.id !== props.id);
+                props.setWatchedList(updatedWatchedList);
+            }
+        }else {
+            if (!props.watchedList.some(movie => movie.id === props.id)) {
+                props.setWatchedList(isWatchedList => [...isWatchedList, {id: props.id, image: props.image, title: props.title}])
+            }
+        }
+    }
+
+    const className = Number(props.rating) < 5 ? 'bad' : Number(props.rating) < 7.5 ? 'okay' : 'good'
```

Comment:
instead of allocating a variable for this, we typically like to do this inline:
```
<div className={{Number(props.rating) < 5 ? 'bad' : Number(props.rating) < 7.5 ? 'okay' : 'good'}}
```